tests/runtime/service_generation/test_create_service.py

```diff
@@ -52,50 +72,54 @@ def test_create_inference_rpcs_for_multiple_modules_of_same_type():
     ]
     rpcs = create_inference_rpcs(module_list)
 
-    # only 2 RPCs, Widget and Gadget because SampleWidget and AnotherWidget are of the same module type Widget
+    # only 2 RPCs, SampleModule and SamplePrimitiveModule have task SampleTask, OtherModule has task OtherTask
     assert len(rpcs) == 2
     assert sample_lib.modules.sample_task.SampleModule in rpcs[0].module_list
     assert sample_lib.modules.sample_task.SamplePrimitiveModule in rpcs[0].module_list
     assert sample_lib.modules.other_task.OtherModule in rpcs[1].module_list
 
 
-def test_create_inference_rpcs_remove_non_primitive_modules():
-    # NOTE: This requires Gadget to be in config since other modules do not have methods - TODO fix?
+def test_create_inference_rpcs_removes_modules_with_no_task():
     module_list = [
-        sample_lib.modules.sample_task.SampleModule,  # is a primitive module
-        sample_lib.modules.sample_task.InnerModule,  # not a primitive module
+        sample_lib.modules.sample_task.SampleModule,  # has a task
+        sample_lib.modules.sample_task.InnerModule,  # does not have a task
     ]
     rpcs = create_inference_rpcs(module_list)
 
-    # only 1 RPC, fidget is not valid
     assert len(rpcs) == 1
     assert sample_lib.modules.sample_task.SampleModule in rpcs[0].module_list
+    assert sample_lib.modules.sample_task.InnerModule not in rpcs[0].module_list
 
 
 ### create_training_rpcs tests #################################################
 
 
-def test_create_inference_rpcs_for_module_with_no_train_function():
-    class Foo:
-        def __init__(self):
+def test_no_training_rpcs_module_with_no_train_function():
+    @caikit.module(
+        id=str(uuid.uuid4()), name="something", version="0.0.0", task=SampleTask
+    )
+    class Foo(caikit.core.ModuleBase):
+        def run(self, sample_input: SampleInputType) -> SampleOutputType:
             pass
 
-        def run(self):
+        def train_in_progress(self):
             pass
 
-        def train_in_progress(self):
+    rpcs = create_training_rpcs([Foo])
+    assert len(rpcs) == 0
+
+
+def test_no_training_rpcs_for_module_with_no_task():
+    @caikit.module(id=str(uuid.uuid4()), name="something", version="0.0.0")
+    class Foo(caikit.core.ModuleBase):
+        def train(self, foo: int) -> "Foo":
             pass
 
-    rpcs = create_inference_rpcs([Foo])
+    rpcs = create_training_rpcs([Foo])
     assert len(rpcs) == 0
 
 
 def test_create_training_rpcs():
     rpcs = create_training_rpcs([widget_class])
     assert len(rpcs) == 1
     assert widget_class in rpcs[0].module_list
-
-
-def test_create_training_rpcs_without_train():
```

Comment:
note for visibility: removing since this would test duplicate functionality as `test_no_training_rpcs_module_with_no_train_function`